src/v2/view-builder/utils/ChallengeViewUtil.js

```diff
@@ -24,7 +24,8 @@ export function doChallenge(view) {
       className: 'loopback-content',
       template: hbs`<div class="spinner"></div>`
     }));
-    view.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest);
+    view.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest,
```

Comment:
can we just pass the object instead of 4 parameters of the same object ?

Reply:
@pradeepdewda-okta Passing a single object would mean we lose the `doLoopback` function default parameters, which arguably is still neater than manually checking each argument for `undefined`.

I considered this during my initial implementation, but decided that despite the verbosity of longer function parameters, the code is still clear. Whereas passing single object entails pushing the parameter checks into the function, which reduces clarity somewhat due to more verbose `undefined` checks.

Reply:
default parameters could be set with in the function and since this function already does a lot, doing parameter checks should also be it's responsibility.. we may want to add 5th argument in future? we should pass the object here instead of params.

Reply:
Both approaches have their own pros/cons. If you are sure Option 2 is the better way forward I will make the changes...

Option 1 (existing):

    doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest,
        deviceChallenge.probeTimeoutMillis);

    function doLoopback(authenticatorDomainUrl = '', ports = [], challengeRequest = '', probeTimeoutMillis = 100) {
      let currentPort;

Option 2:

    doLoopback(deviceChallenge);

    function doLoopback(deviceChallenge) {
      let authenticatorDomainUrl = deviceChallenge.domain !== undefined ? deviceChallenge.domain : '';
      let ports = deviceChallenge.ports !== undefined ? deviceChallenge.ports : []
      let challengeRequest = deviceChallenge.challengeRequest !== undefined ? deviceChallenge.challengeRequest : ''
      let probeTimeoutMillis = deviceChallenge.probeTimeoutMillis !== undefined ? deviceChallenge.probeTimeoutMillis : 100
      let currentPort;

Reply:
thanks @howardlee-okta this is what i had in mind.. #option2

Reply:
Thanks @pradeepdewda-okta - all done. Can you approve/merge?